<?xml version="1.0"?>
<launch>

  <include file="$(find freenect_launch)/launch/freenect.launch">
    <arg name="depth_registration" value="true" />
  </include>
  <node name="depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan_node" output="screen">
    <remap from="input/image" to="/camera/depth/image_raw" />
    <remap from="output/scan" to="/scan" />
  </node>

  <include file="$(find rtabmap_ros)/launch/rgbd_mapping.launch">
    <arg name="rtabmap_args" value="--delete_db_on_start --Vis/MaxFeatures 500 --Mem/ImagePreDecimation 2 --Mem/ImagePostDecimation 2 --Kp/DetectorStrategy 6 --OdomF2M/MaxSize 1000 --Odom/ImageDecimation 2" />
    <arg name="rtabmapviz" value="false" />
  </include>
  
  <rosparam file="$(find autonomus_mobile_robot)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find autonomus_mobile_robot)/config/joint_limits.yaml" command="load"/>

  <arg name="model" default="$(find autonomus_mobile_robot)/urdf/mobile_robot.urdf.xacro"/>
  
  <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
  
  <!-- <node name="robot_hardware_interface" pkg="autonomus_mobile_robot" type="mobile_robot_hardware_interface" output="screen">
    <remap from="/mobile_robot/mobile_base_controller/cmd_vel" to="/cmd_vel"/>
   </node> -->

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
  </node>
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find autonomus_mobile_robot)/config/rviz_config_urdf_test.rviz"/>
  
  <!-- <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
            /mobile_robot/joints_update
            /mobile_robot/mobile_base_controller" >
  </node> -->

  <node name="telop" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py">
		<!--<remap from="/cmd_vel" to="/mobile_robot/mobile_base_controller/cmd_vel"/> -->
  </node>

  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <param name="map_update_interval" value="5.0"/>
	<param name="maxUrange" value="6.0"/>
    <param name="maxRange" value="8.0"/>
    <param name="linearUpdate" value="0.05"/>
	<param name="angularUpdate" value="0.5"/>
  </node>
   
</launch>
